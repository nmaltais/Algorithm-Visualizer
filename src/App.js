import React, { Component } from 'react';
import './App.css';
import 'semantic-ui-css/semantic.min.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Grid, Sidebar } from 'semantic-ui-react';
import Navbar from './components/Navbar';
import Pathfinding from './components/Pathfinding';
import SortingClassComponent from './components/sorting/SortingClassComponent';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { visible: true };
  }

  toggleSidebar = () => this.setState((prevState) => ({ visible: !prevState.visible }))

  render() {
    const { visible } = this.state;
    return (
      <Router>
        <Grid style={{ margin: '0px' }}>
          <Grid.Column width={16} style={{ minHeight: '100vh', padding: '0px' }}>
            <Sidebar.Pushable>
              <Navbar visible={visible} toggleSidebar={this.toggleSidebar} />
              <Sidebar.Pusher>
                <Switch>
                  <Route path="/sorting/:alg">
                    <SortingClassComponent />
                  </Route>
                  <Route path="/pathfinding">
                    <Pathfinding />
                  </Route>
                  <Route path="/">
                    <p>Home Page</p>
                  </Route>
                </Switch>
              </Sidebar.Pusher>
            </Sidebar.Pushable>
          </Grid.Column>
        </Grid>
      </Router>
    );
  }
}

export default App;
