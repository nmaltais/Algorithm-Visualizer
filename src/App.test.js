import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('renders correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('toggleSidebar toggles sidebar visibility in App state', () => {
  // Setup
  const wrapper = shallow(<App />);

  expect(wrapper.state('visible')).toEqual(true);
  wrapper.instance().toggleSidebar('push');
  expect(wrapper.state('visible')).toEqual(false);
  wrapper.instance().toggleSidebar('push');
  expect(wrapper.state('visible')).toEqual(true);
});
