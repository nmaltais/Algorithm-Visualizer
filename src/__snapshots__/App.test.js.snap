// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Grid
    style={
      Object {
        "margin": "0px",
      }
    }
  >
    <GridColumn
      style={
        Object {
          "minHeight": "100vh",
          "padding": "0px",
        }
      }
      width={16}
    >
      <SidebarPushable>
        <Navbar
          toggleSidebar={[Function]}
          visible={true}
        />
        <SidebarPusher>
          <Switch>
            <Route
              path="/sorting"
            >
              <Sorting
                toggleSidebar={[Function]}
              />
            </Route>
            <Route
              path="/pathfinding"
            >
              <Pathfinding />
            </Route>
            <Route
              path="/"
            >
              <p>
                Home Page
              </p>
            </Route>
          </Switch>
        </SidebarPusher>
      </SidebarPushable>
    </GridColumn>
  </Grid>
</BrowserRouter>
`;
